mk/protobuf.mk

```diff
@@ -0,0 +1,83 @@
+# Copyright 2014 The Cockroach Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License. See the AUTHORS file
+# for names of contributors.
+#
+# Author: Tamir Duberstein (tamird@gmail.com)
+
+REPO_ROOT      := .
+ORG_ROOT       := $(REPO_ROOT)/..
+GITHUB_ROOT    := $(ORG_ROOT)/..
+GOGOPROTO_ROOT := $(GITHUB_ROOT)/gogo/protobuf
+
+ENGINE_ROOT := $(REPO_ROOT)/storage/engine
+
+# Ensure we have an unambiguous GOPATH
+RELATIVE_GOPATH := $(GITHUB_ROOT)/../..
+#                                 ^  ^~ GOPATH
+#                                 |~ GOPATH/src
+
+GOPATH_BIN      := $(RELATIVE_GOPATH)/bin
+PROTOC          := $(GOPATH_BIN)/protoc
+PROTOC_PLUGIN   := $(GOPATH_BIN)/protoc-gen-gogoroach
+GOGO_PROTO      := $(GOGOPROTO_ROOT)/gogoproto/gogo.proto
+GOGO_PROTO_PATH := $(GOGOPROTO_ROOT):$(GOGOPROTO_ROOT)/protobuf
+
+GOTOOLS := .go_tools
+CPPTOOLS := .cpp_tools
+
+GO_PROTOS  := $(sort $(shell find . -name *.proto -type f))
```

Comment:
You should probably avoid searching in the `.git` directory. Might be better to use `git ls-tree`, though that wouldn't find files that haven't been added to git.